internal/keyrotation/rotate.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package keyrotation implements the API handlers for running key rotation jobs.
+package keyrotation
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/logging"
+	"go.opencensus.io/trace"
+)
+
+func (s *Server) handleRotateKeys(ctx context.Context) http.HandlerFunc {
+	logger := logging.FromContext(ctx).Named("keyrotation.HandleRotate")
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		_, span := trace.StartSpan(r.Context(), "(*keyrotation.handler).ServeHTTP")
+		defer span.End()
+
+		s.mu.Lock()
```

Comment:
I guess we can cheat and set the configuration for this service to be `maxInstances=1`, but we should consider a database-level lock or something.

Reply:
yeah - db level lock is preferred